### PR TITLE
Filter trackers from measurements and fixed empty aggregated 

### PR DESCRIPTION
**Pull request by Casper-NS**
Filter trackers mac adresses, so they are not part of the measurements.
Fixed an issue with empty measurements not being stored properly.
Added some more tests.

### CODE REVIEW

File: CentralHub.Api/Controllers/MeasurementController.cs [hunk 1]
```diff
@@ -15,21 +15,38 @@ public sealed class MeasurementController : ControllerBase
...
+        ITrackerRepository trackerRepository)
     {
         _logger = logger;
         _aggregatorRepository = aggregatorRepository;
+        _trackerRepository = trackerRepository;
     }
 
     [HttpPost("add")]
-    public async Task AddMeasurements(AddMeasurementsRequest addMeasurementsRequest, CancellationToken token)
+    public async Task<AddMeasurementResponse> AddMeasurements(AddMeasurementsRequest addMeasurementsRequest, CancellationToken token)
```
Comment (Taoshix) [inline CentralHub.Api/Controllers/MeasurementController.cs:31]: Async task without async in the name, me not understand why

Comment (Casper-NS) [reply CentralHub.Api/Controllers/MeasurementController.cs:31]: idk, its like the others...

**Review by Taoshix — APPROVED** [review]
We need this filtering ASAP

File: CentralHub.Api.Model/Responses/Measurements/AddMeasurementResponse.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+using System.Text.Json.Serialization;
+
+namespace CentralHub.Api.Model.Responses.AggregatedMeasurements;
+
+public sealed class AddMeasurementResponse
```
Comment (mads256h) [inline CentralHub.Api.Model/Responses/Measurements/AddMeasurementResponse.cs:5]:
```suggestion
public sealed class AddMeasurementsResponse
```

File: CentralHub.Api.Tests/MeasurementControllerTests.cs [hunk 3]
```diff
@@ -63,16 +64,31 @@ public void TearDown()
...
-        var addedMeasurements = (await _aggregatedMeasurementRepository.GetTrackerMeasurementGroupsAsync(default))[_trackerId][0].Measurements;
+        var addedMeasurements = (await _aggregatedMeasurementRepository.GetRoomMeasurementGroupsAsync(default))[_trackerId][0].Measurements;
 
         Assert.That(addedMeasurements, Has.Count.EqualTo(measurements.Length));
         Assert.That(addedMeasurements, Does.Contain(measurements[0]));
         Assert.That(addedMeasurements, Does.Contain(measurements[1]));
     }
+
+    [Test]
+    public async Task TestAddTrackerAsMeasurements()
```
Comment (mads256h) [inline CentralHub.Api.Tests/MeasurementControllerTests.cs:81]:
```suggestion
    public async Task TrackersFilteredFromMeasurements
```

File: CentralHub.Api.Tests/MockTrackerRepository.cs [hunk 4]
```diff
@@ -86,4 +86,9 @@ public Task AddUnregisteredTracker(string wifiMacAddress, string bluetoothMacAdd
             });
         return Task.CompletedTask;
     }
+
+    public Task<IEnumerable<TrackerDto>> GetRegisteredTrackers(CancellationToken cancellationToken)
+    {
+        return Task.FromResult(RoomDto.Trackers.AsEnumerable());
```
Comment (mads256h) [inline CentralHub.Api.Tests/MockTrackerRepository.cs:92]: Make a copy

File: CentralHub.Api/Controllers/MeasurementController.cs [hunk 5]
```diff
@@ -15,21 +15,40 @@ public sealed class MeasurementController : ControllerBase
...
-        await _aggregatorRepository.AddMeasurementsAsync(
-            addMeasurementsRequest.TrackerId,
-            addMeasurementsRequest.Measurements,
-            token);
+        var registeredTrackers = await _trackerRepository.GetRegisteredTrackers(token);
+        var unregisteredTrackers = await _trackerRepository.GetUnregisteredTrackers(token);
+
+        var tracker = registeredTrackers.SingleOrDefault(t => t.TrackerDtoId == addMeasurementsRequest.TrackerId);
+
+        if (tracker != null)
```
Comment (mads256h) [inline CentralHub.Api/Controllers/MeasurementController.cs:38]: Invert this if and return AddMeasurementResponse.CreateUnsuccessful() to reduce nesting

File: CentralHub.Api/Services/LocalizationService.cs [hunk 6]
```diff
@@ -40,51 +37,53 @@ public async Task DoWorkAsync(CancellationToken stoppingToken)
...
     {
         var measurements = measurementGroups.SelectMany(mg => mg.Measurements);
         var bluetooth = measurements.Where(m => m.Type == Measurement.Protocol.Bluetooth);
         var wifi = measurements.Where(m => m.Type == Measurement.Protocol.Wifi);
+        var now = DateTime.Now;
 
-        var startMeasurementTime = measurementGroups.Min(mg => mg.Timestamp);
-        var endMeasurementTime = measurementGroups.Max(mg => mg.Timestamp);
+        var startMeasurementTime = measurementGroups.Count != 0 ? measurementGroups.Min(mg => mg.Timestamp) : now - _sleepTime;
+        var endMeasurementTime = measurementGroups.Count != 0 ? measurementGroups.Max(mg => mg.Timestamp) : now;
```
Comment (mads256h) [inline CentralHub.Api/Services/LocalizationService.cs:79]:
```suggestion
        var startMeasurementTime = measurementGroups.Any() ? measurementGroups.Min(mg => mg.Timestamp) : now - _sleepTime;
        var endMeasurementTime = measurementGroups.Any() ? measurementGroups.Max(mg => mg.Timestamp) : now;
```

**Review by mads256h — CHANGES_REQUESTED** [review]

File: CentralHub.Api.Tests/MockTrackerRepository.cs (hunk 4 above)
Comment (Casper-NS) [reply CentralHub.Api.Tests/MockTrackerRepository.cs]: fixed

**Pull request merged**